### ISSUE TITLE
Is SUSHI Slice Ordering Arbitrary?

### ISSUE THREAD

**Issue by cmoesel** [FHIR/sushi#1029]
The use case in #1028 also raises another issue.  When the Questionnaire instance is modified to use slice names instead of indices, the slice names seem to come out in an arbitrary order.

E.g., the profile defines the slices in this order:
```
* item.item contains question 1..1 MS and currentValue 1..1 MS and answerSuggestion 1..1 MS
```

The instance defines them in the same order:
```
* item[=].item[question].linkId = "a_question"
// more question rules
* item[=].item[currentValue].linkId = "a_current_value"
// more currentValue rules
* item[=].item[answerSuggestion].linkId = "a_answer_suggestion"
// more answerSuggestion rules
```
(FSH Online: https://fshschool.org/FSHOnline/#/share/3Hljd7Y)

But when SUSHI generates the instance JSON, it comes out in the order: `currentValue`, `answerSuggestion`, `question`:
```json5
{
  "resourceType": "Questionnaire",
  // other properties omitted for brevity
  "item": [
    {
      "item": [
        {
          "text": "Current Value"
          // other properties omitted for brevity
        },
        {
          "text": "Answer Suggestion"
          // other properties omitted for brevity
        },
        {
          "text": "This is the Question"
          // other properties omitted for brevity
        }
      ]
     // other properties omitted for brevity
    }
  ],
  // other properties omitted for brevity
}
```
This order doesn't match the order in which the slices were defined nor the order in which the elements were defined in the instance.  I'd expect it to match at least one of those.  Based on SUSHI's current approach of always putting implied required elements first, I'd expect it to match the order the slices were defined in the profile.

We should look into this to determine if SUSHI is just ordering things arbitrarily or if there is a method to the madness.  Either way, we need to be careful about changing order now since some people may be relying on this specific order.

**Comment by mint-thompson** [comment]
Something I did not address in my previous comment is cases where both names and numeric indices are used. Right now, this will result in... problems. Please don't do this. Please just use slice names. It's easier for everyone involved.

**Comment by jafeltra** [comment]
Resolved with SUSHI 3.0.0 updates for slice ordering
